modules/core/core-scheduler/src/main/java/com/enonic/xp/impl/scheduler/distributed/RescheduleTask.java

```diff
@@ -0,0 +1,114 @@
+package com.enonic.xp.impl.scheduler.distributed;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.HazelcastInstanceAware;
+import com.hazelcast.scheduledexecutor.DuplicateTaskException;
+import com.hazelcast.scheduledexecutor.IScheduledExecutorService;
+import com.hazelcast.scheduledexecutor.NamedTask;
+
+import com.enonic.xp.context.Context;
+import com.enonic.xp.context.ContextAccessor;
+import com.enonic.xp.context.ContextBuilder;
+import com.enonic.xp.core.internal.osgi.OsgiSupport;
+import com.enonic.xp.scheduler.ScheduledJob;
+import com.enonic.xp.scheduler.SchedulerName;
+import com.enonic.xp.scheduler.SchedulerService;
+import com.enonic.xp.security.RoleKeys;
+import com.enonic.xp.security.auth.AuthenticationInfo;
+
+public class RescheduleTask
+    implements NamedTask, HazelcastInstanceAware, Runnable, Serializable
+{
+    private static final long serialVersionUID = 0;
+
+    private static final Logger LOG = LoggerFactory.getLogger( RescheduleTask.class );
+
+    private transient IScheduledExecutorService schedulerExecutorService;
+
+    @Override
+    public String getName()
+    {
+        return "rescheduleTask";
+    }
+
+    @Override
+    public void setHazelcastInstance( final HazelcastInstance hazelcastInstance )
+    {
+        this.schedulerExecutorService = hazelcastInstance.getScheduledExecutorService( "scheduler" );
+    }
+
+    @Override
+    public void run()
+    {
+        try
+        {
+            this.doRun();
+        }
+        catch ( Exception e )
+        {
+            LOG.warn( "Problem during task scheduling.", e );
```

Comment:
that is an error, not warning. And no need to log either, just propagate?

Reply:
we don't need to stop RescheduleTask if it failed on any exception. Any task can't be rerun in this case.